Dockerfile

```diff
@@ -1,8 +1,42 @@
-FROM python:3.6
-RUN set -x && \
-	pip install nornir
-VOLUME [ "/nornir" ]
-WORKDIR "/nornir"
-ENTRYPOINT [ "/usr/local/bin/python" ]
-#usage:
-# from inside your nornir base directory run: docker run -it --rm -v `pwd`:/nornir nornir:latest get_facts.py
+ARG PYTHON
+# Stage 1 - Build and install dependencies
+FROM python:${PYTHON}-slim-stretch AS builder
+
+WORKDIR /nornir
+ENV PATH="/root/.poetry/bin:$PATH"
+
+RUN apt-get update \
+    && apt-get install -yq curl git \
+    && curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py | python \
+    && python -m venv .venv \
+    && poetry config settings.virtualenvs.in-project true \
```

Comment:
shouldn't this be `settings.virtualenvs.create false`?

Reply:
`settings.virtualenvs.create false` this will use system python
`settings.virtualenvs.in-project true` this will use `.venv` inside project dir. The beauty of the latter is that we can now move the whole project folder around (e.g. between multistage)

Reply:
Why do you want to create a virtualenv inside a container?

Reply:
I guess it's my personal preference to keep venvs in docker, do you want me to change it?

Reply:
Yeah, I'd like to remove things that are not necessary to keep it simple

Reply:
I've removed multi-stage build as agreed